builder/vsphere/clone/step_customize.go

```diff
@@ -273,4 +302,135 @@ func (s *StepCustomize) globalIpSettings() types.CustomizationGlobalIPSettings {
 	}
 }
 
+func (l *LinuxOptions) prepare(errs []error) []error {
+	if l.Hostname == "" {
+		errs = append(errs, fmt.Errorf("linux options `host_name` is empty"))
+	}
+	if l.Domain == "" {
+		errs = append(errs, fmt.Errorf("linux options `domain` is empty"))
+	}
+
+	if l.HWClockUTC == config.TriUnset {
+		l.HWClockUTC = config.TriTrue
+	}
+	if l.Timezone == "" {
+		l.Timezone = "UTC"
+	}
+	return errs
+}
+
+func (l *LinuxOptions) linuxPrep() *types.CustomizationLinuxPrep {
+	obj := &types.CustomizationLinuxPrep{
+		HostName: &types.CustomizationFixedName{
+			Name: l.Hostname,
+		},
+		Domain:     l.Domain,
+		TimeZone:   l.Timezone,
+		HwClockUTC: l.HWClockUTC.ToBoolPointer(),
+	}
+	return obj
+}
+
+func (w *WindowsOptions) prepare(errs []error) []error {
+	if w.ComputerName == "" {
+		errs = append(errs, fmt.Errorf("The `computer_name` is required"))
+	}
+	if w.Workgroup != "" {
+		if w.JoinDomain != "" {
+			errs = append(errs, fmt.Errorf("The `workgroup` and `join_domain` must not be set together"))
+		}
+	}
+	if w.JoinDomain != "" {
+		if w.DomainAdminUser == "" {
+			errs = append(errs, fmt.Errorf("The `domain_admin_user` is required when `join_domain` is specified"))
+		}
+		if w.DomainAdminPassword == "" {
+			errs = append(errs, fmt.Errorf("The `domain_admin_password` is required when `join_domain` is specified"))
+		}
+	}
```

Comment:
I don't think it makes good sense to include the ability to join the machine image to a domain per comments in #34.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

Reply:
ok removed.